### ISSUE TITLE
[Vegas] Incorrect score

### ISSUE THREAD

**Issue by sbe15** [TobiasBielefeld/Simple-Solitaire#63]
When moving an Ace to the foundation, score will be increased, however it will also increase if the Ace was moved _from_ foundation _to_ foundation. That way, score can be increased indefinetly by just moving around one Ace within the foundation.

My guess is that something is wrong in [Simple-Solitaire/app/src/main/java/de/tobiasbielefeld/solitaire/games/Vegas.java](Simple-Solitaire/app/src/main/java/de/tobiasbielefeld/solitaire/games/Vegas.java):
```
public int addPointsToScore(ArrayList<Card> cards, int[] originIDs, int[] destinationIDs, boolean isUndoMovement) {
        int originID = originIDs[0];
        int destinationID = destinationIDs[0];

        if (destinationID >= 7 && destinationID <= 10){ /* <= should check originID as well */
            return betAmount/10;
        }

        if (originID >= 7 && originID <= 10){
            return -2*betAmount/10;
        }

        return 0;
}
```

**Comment by TobiasBielefeld** [comment]
Thanks for noticing this issue! You are right, i needed to check the originID as well. I also found issues related to the deal3 option (undo moving cards from waste to foundations didn't undo the score as well).

This should be fixed in version 3.8.3

**Comment by sbe15** [comment]
Just updated and it works fine now
Thanks :+1:
